dm/master/scheduler/scheduler.go

```diff
@@ -1584,6 +1595,13 @@ func (s *Scheduler) tryBoundForWorker(w *Worker) (bounded bool, err error) {
 // called should update the s.unbounds.
 func (s *Scheduler) tryBoundForSource(source string) (bool, error) {
 	var worker *Worker
+	// 0. check whether this source has source configuration. If not, we shouldn't bound this but just abandon it
+	// This usually happens when dm didn't handle the source compatibility correctly. For example, downgrade from a higher dm version
+	// We can just mark this source bounded to abandon this source. After we fix the compatibility problem dm-master will restart the scheduler or manually add this source
+	if _, ok := s.sourceCfgs[source]; !ok {
+		s.logger.Warn("source configuration not found for the source to bound, add this source to unbounded", zap.String("source", source))
+		return true, nil
```

Comment:
maybe 
a) we should `return false, nil`, to inform caller add this source to unbound. or
b) we maintain the invariant: "after scheduler is started, all sources in `bound`, `unbound` must have their source config, sources in `lastBound` may not have source configs". So if `tryBoundForSource` is not called for `lastBound`, this check could be deleted.

I prefer b) @GMHDBJD

Reply:
b) LGTM